utensor_cgen/backend/base.py

```diff
@@ -16,15 +16,33 @@ def __new__(cls, config, *args, **kwargs):
         validator(config, *args, **kwargs)
     if isinstance(config, dict):
       config = Configuration(cls.default_config, config)
+    elif config is None:
+      config = Configuration(cls.default_config, {})
     self._config = config
     return self
 
+  @abstractmethod
   def apply(self, ugraph):
```

Comment:
You mean sth like this: `register_visit_op(op_type, callback)`.
So user can register a callback to the backend whenever some specific op_type is visited in the body of `apply`, the callback will be invoked?

And how does tensor visitor works? register a callback on tensor of specific data type?
Since tensor itself is kinda generic. It's functionality is mostly given by the op which owns it.

Reply:
https://github.com/uTensor/utensor_cgen/issues/106